rpmautospec/pkg_history.py

```diff
@@ -123,15 +123,29 @@ def _get_rpmverflags(cls, path: str, name: Optional[str] = None) -> Optional[str
             "--qf",
             query,
             "--specfile",
-            f"{name}.spec",
-        ]
-
-        call = subprocess.run(
-            rpm_cmd,
-            cwd=path,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
         )
+
+        with specfile.open(mode="rb") as unabridged, NamedTemporaryFile(
+            mode="wb", prefix=f"rpmautospec-abridged-{name}-", suffix=".spec", delete=False
+        ) as abridged:
+            # Attempt to parse a shortened version of the spec file first, to speed up processing in
+            # certain cases.
+            for line in unabridged:
+                if line.strip() == b"%prep":
+                    break
+                abridged.write(line)
+            abridged.flush()
```

Comment:
For future maintainers, please make the comments more verbose about why we're splitting on the `%prep` section header.